File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        int l=0;
        int r=nums.size() -1;

        while(l<r){
            int curSum = nums[l] + nums[r];
            if(curSum<target){
                l++;
            }else if(curSum>target){
                r--;
            }else{
                return {l+1 , r+1};
            }
        }
    }

};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        res = defaultdict(int)

        for num in nums:
            res[num] += 1        
        
        sol = sorted(res, key=res.get, reverse=True)
        
        n = []
        i = 0
        while i < k:
            n.append(sol[i])
            i += 1
        return n